ecosystem_cicd_tools/new_cicd/actions.py

```diff
@@ -45,6 +48,18 @@ def upload_assets_to_release(assets, release_name, repository, **_):
             os.environ.get('CIRCLE_USERNAME', 'earthmant')
         )
 
+    time.sleep(180)
+
+    name_plugin = repository.name
+    plugin_id = get_plugin_id_from_marketplace(name_plugin)
+    version = get_plugin_versions_from_marketplace(plugin_id)[-1]
+    node_types = get_node_types_for_plugin_version(name_plugin,
+                                                   version)
+
+    if release_name not in version and not node_types:
+        raise RuntimeError(
+            'Failed to update marketplace with plugin release.')
+
```

Comment:
Instead of putting all of the verification logic here, create a separate function `all_plugin_stuff_is_in_marketplace` and it should call the new functions that I mentioned in the first comment.